homeassistant/components/tuya/climate.py

```diff
@@ -226,6 +231,12 @@ def __init__(
                 description.switch_only_hvac_mode,
             ]
 
+        # Determine dpcode to use for checking thermostat current operation
+        if (self._is_thermostat) and self.find_dpcode(
+            DPCode.VALVE_STATE, dptype=DPType.ENUM
+        ):
+            self._attr_hvac_action = HVACAction.OFF
```

Comment:
This is setting a state, which should not be part of the init?

Reply:
I've never deal with such kind of code before, therefore I may made some mistakes. My idea here was to use that initialization as a flag, that will later be used in hvac_action property, since self._attr_hvac_action is initialized as None in Climate class

Reply:
>  My idea here was to use that initialization as a flag

It should not have to be initialized, as the state is updated on adding to Home Assistant.

Reply:
This code is removed